src/pipeline/src/manager/table.rs

```diff
@@ -325,24 +302,103 @@ impl PipelineTable {
 
         {
             self.pipelines.insert(
-                Self::generate_pipeline_cache_key(schema, name, None),
+                generate_pipeline_cache_key(schema, name, None),
                 compiled_pipeline.clone(),
             );
             self.pipelines.insert(
-                Self::generate_pipeline_cache_key(schema, name, Some(TimestampNanosecond(version))),
+                generate_pipeline_cache_key(schema, name, Some(TimestampNanosecond(version))),
                 compiled_pipeline.clone(),
             );
         }
 
         Ok((version, compiled_pipeline))
     }
 
-    async fn find_pipeline_by_name(
+    pub async fn delete_pipeline(
+        &self,
+        schema: &str,
+        name: &str,
+        version: PipelineVersion,
+    ) -> Result<Option<()>> {
+        // 0. version is ensured at the http api level not None
+        ensure!(
+            version.is_some(),
+            InvalidPipelineVersionSnafu { version: "None" }
+        );
+
+        // 1. check pipeline exist in catalog
+        let pipeline = self.find_pipeline(schema, name, version).await?;
+        if pipeline.is_none() {
+            return Ok(None);
+        }
+
+        // 2. do delete
+        let table_info = self.table.table_info();
```

Comment:
Can we just use a Sqlhandler here to execute a SQL? I feel this part is touching internals of the query engine which is a little more detailed for this layer of abstraction.

Reply:
I think we can do this. Let's merge this pr first, and I will create a new issue/pr later.